### PR TITLE
fixed invalid url old timerseries

### PR DESCRIPTION
**Pull request by Shrhawk**
**Checklist**:

* [ ] All tests are passing
* [ ] I have added any necessary documentation and/or release notes
* [ ] I have followed the [commit message guidelines](https://wiki.antonischristofides.com/en/sops/production/development/git-and-github#committing-and-commit-messages) and the ["before submitting a pull request" guidelines](https://wiki.antonischristofides.com/en/sops/production/development/git-and-github#before-submitting-a-pull-request)
* [ ] If I introduced (or abolished) any third-party library, I will write a short paragraph in this PR [according to the guidelines](https://wiki.antonischristofides.com/sops/production/development/third-party-libraries)
* [ ] I have removed all obsoleted code

### CODE REVIEW

**Comment by Shrhawk** [comment]
https://github.com/openmeteo/enhydris/issues/493

File: enhydris/api/views.py [hunk 1]
```diff
@@ -237,7 +238,7 @@ def data(self, request, pk=None, *, station_id, timeseries_group_id=None):
 
     @action(detail=True, methods=["get"])
     def bottom(self, request, pk=None, *, station_id, timeseries_group_id=None):
-        ts = get_object_or_404(models.Timeseries, pk=pk)
+        ts = get_object_or_404(models.Timeseries, pk=int(pk))
```
Comment (aptiko) [inline enhydris/api/views.py:241]: Was this working before? If yes, why is this change needed? (And how on earth was it working?)

File: enhydris/api/views.py [hunk 2]
```diff
@@ -246,7 +247,7 @@ def bottom(self, request, pk=None, *, station_id, timeseries_group_id=None):
 
     @action(detail=True, methods=["get"])
     def chart(self, request, pk=None, *, station_id, timeseries_group_id=None):
-        timeseries = get_object_or_404(models.Timeseries, pk=pk)
+        timeseries = get_object_or_404(models.Timeseries, pk=int(pk))
```
Comment (aptiko) [inline enhydris/api/views.py:250]: Same question.

File: enhydris/api/views.py [hunk 3]
```diff
@@ -167,6 +167,7 @@ class TimeseriesViewSet(ModelViewSet):
     CHART_MAX_INTERVALS = 200
     queryset = models.Timeseries.objects.all()
     serializer_class = serializers.TimeseriesSerializer
+    lookup_value_regex = '\d+'
```
Comment (aptiko) [inline enhydris/api/views.py:170]: If this line is added, then I can't reproduce the problem any more. If that is the case, why do we have the `try...except` in `_get_data()`?

**Review by aptiko — CHANGES_REQUESTED** [review]
1. The commit message needs proper capitalization and spelling, and should also be in the imperative. It should also have `(fixes #493)` in the end. (See also "Git and GitHub" [section "Commit and commit messages"] in Confluence.)
2. I found out that this problem also occurs in the new URLs after all, not only in the deprecated; for example, under `/api/stations/2012/timeseriesgroups/584/timeseries/{hello}/data/`. This means the commit message should not mention the "old" URL. It could be, for example, `Fix crash in invalid timeseries URL (fixes #493)`.

File: enhydris/api/views.py (hunk 1 above)
Comment (Shrhawk) [reply enhydris/api/views.py:241]: Yes @aptiko it was working fine, just make it consistent for the views

File: enhydris/api/views.py (hunk 3 above)
Comment (Shrhawk) [reply enhydris/api/views.py:170]: @aptiko added try catch at `_get_data()` here bcz it's not the direct part of view, can be called anywhere from the code

**Comment by Shrhawk** [comment]
Closed in favor of https://github.com/openmeteo/enhydris/pull/495 (renamed the branch)

**Pull request closed without merging**